x-pack/plugins/alerting/server/data/rule/types/rule_attributes.ts

```diff
@@ -41,12 +41,12 @@ export interface RRuleAttributes {
   byweekday?: Array<string | number>;
   bymonth?: number[];
   bysetpos?: number[];
-  bymonthday: number[];
-  byyearday: number[];
-  byweekno: number[];
-  byhour: number[];
-  byminute: number[];
-  bysecond: number[];
+  bymonthday?: number[];
+  byyearday?: number[];
+  byweekno?: number[];
+  byhour?: number[];
+  byminute?: number[];
+  bysecond?: number[];
```

Comment:
I really do not like that. why we are changing the type here

Reply:
@XavierM  Types were broken after merge with the main because of this PR: https://github.com/elastic/kibana/pull/168886/files. And that how I manage to fix our types.